src/test/scala/io/opentargets/etl/backend/MoleculeTest.scala

```diff
@@ -0,0 +1,296 @@
+package io.opentargets.etl.backend
+
+import io.opentargets.etl.backend.MoleculeTest.{getMoleculeInstance, getSampleHierarchyData, getSampleSynonymData}
+import io.opentargets.etl.backend.drug_beta.Molecule
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.{DataFrame, Dataset, Row, SparkSession}
+import org.apache.spark.sql.types.{ArrayType, BooleanType, LongType, MapType, StringType, StructField, StructType}
+import org.scalatest.PrivateMethodTester
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.must.Matchers
+
+object MoleculeTest {
```

Comment:
honestly, I love seeing tests. thanks, @JarrodBaker for taking the time to include tests.